holidays/holiday_base.py

```diff
@@ -633,6 +642,12 @@ def _populate(self, year: int) -> Set[Optional[date]]:
 
         return dates
 
+    def _populate_categories(self):
```

Comment:
The method names for subdiv and category holidays need to be aligned, e.g., either `_add_subdiv_holidays` and `_add_categories_holidays` or `_populate_subdiv_holidays` and `_populate_categories_holidays`. 

I'd also put both of them after special holidays part if possible.

Reply:
I considered "populate" to be something more _global_ than "add", so I named it that way. BTW, need to implement categories support for subdivisions too 🙄

Reply:
In the end both of them do the same stuff -- update the internal dict with new items. Their only difference to me is the update context: category vs subdiv.
I'm sure it's okay to keep the names as is within this PR.